code/modules/security levels/keycard authentication.dm

```diff
@@ -169,6 +169,10 @@ var/global/maint_all_access = 0
 	security_announcement.Announce("The maintenance access requirement has been readded on all maintenance airlocks.","Attention!")
 
 /obj/machinery/door/airlock/allowed(mob/M)
-	if(maint_all_access && src.check_access_list(list(access_maint_tunnels)))
+	var/obj/item/id = M.GetIdCard()
```

Comment:
```suggestion
	var/obj/item/id/id = M.GetIdCard()
```
Be careful with these typecasts. Without this edit, the object you defined is of type `/obj/item`, containing an `/obj/item/id`, with `id` as the variable name.

Reply:
So, as far as I can tell, the ID card variable type is obj/item/card/id, which would make this line fine. Additionally, the GetIdCard() proc returns a obj/item, so this seems correct. I have renamed the variable to obj/item/I to avoid confusion however.